src/Sets/HParallelotope.jl

```diff
@@ -403,3 +404,25 @@ function volume(P::HParallelotope)
     n = dim(P)
     return 2^n * abs(det(genmat(P)))
 end
+
+"""
+    vertices_list(P::HParallelotope)
+
+Return a list of the vertices of a parallelotope in constraint representation.
+
+### Input
+
+- `P` -- parallelotope in constraint representation
+
+### Output
+
+A list of the vertices.
+
+### Algorithm
+
+We call `vertices_list` for an `HPolyhedron`. We do not assume `HPolytope`
+because the set may be wrongly represented and in fact be unbounded.
+"""
+function vertices_list(P::HParallelotope)
```

Comment:
Why is 

```julia
julia> H = [1.0 1.0; 0.0 1.0];
julia> o = [2.5, 2.0, -4.0, -1.0];
julia> Pl = HParallelotope(H, o);  # this set is empty

julia> vertices_list(Pl)
4-element Vector{Vector{Float64}}:
 [2.0, 2.0]
 [0.5, 2.0]
 [1.5, 1.0]
 [3.0, 1.0]
```

returning an incorrect result in the first place?

Reply:
The code calls this method, which uses the generator representation:

https://github.com/JuliaReach/LazySets.jl/blob/b11803321cd10ce904198d8abcf19759f9257f1d/src/Sets/Zonotope.jl#L516-L528

It seems that the generator representation is incorrect for empty sets. But it is unclear what the generators should be. Maybe we should make `HParallelotope` not a subtype of `AbstractZonotope` anymore (zonotopes cannot be empty)? Or should we check for emptiness in the constructor (could be a user flag)?

Reply:
Arguably the second option is better; or we can also check for emptiness of `HParallelotope` inside its `vertices_list` function, and if it is non-empty, then call the generator representation implementation.

Reply:
Okay, I think I will go with the last option. So the plan is to change this PR to add an emptiness check to `vertices_list` and in the non-empty case fall back to the `AbstractZonotope` implementation as before.

Reply:
I just realized that `isempty` also converts to `HPolyhedron`, so that would not be cheaper. And if the parallelotope is unbounded, the result would still be wrong (another ill definition one can unfortunately make).

Given that, I changed my mind and suggest again that we either add a check in the constructor for emptiness and boundedness (which we can make controllable by a flag) or remove the assumption that an `HParallelotope` is zonotopic.

Reply:
Adding a check in the constructor sounds good to me.